plugins/localfile-securitymodule/src/test/java/org/hyperledger/besu/plugin/services/securitymodule/localfile/LocalFileSecurityModulePluginTest.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.services.securitymodule.localfile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.hyperledger.besu.plugin.services.securitymodule.localfile.LocalFileSecurityModulePlugin.PICOCLI_NAMESPACE;
+import static org.hyperledger.besu.plugin.services.securitymodule.localfile.LocalFileSecurityModulePlugin.SECURITY_MODULE_NAME;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+import static org.mockito.Mockito.when;
+
+import org.hyperledger.besu.crypto.KeyPairUtil;
+import org.hyperledger.besu.crypto.SECP256K1;
+import org.hyperledger.besu.plugin.BesuContext;
+import org.hyperledger.besu.plugin.services.PicoCLIOptions;
+import org.hyperledger.besu.plugin.services.SecurityModuleService;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModule;
+import org.hyperledger.besu.plugin.services.securitymodule.localfile.configuration.LocalFileSecurityModuleCLIOptions;
+import org.hyperledger.besu.services.BesuConfigurationImpl;
+import org.hyperledger.besu.services.PicoCLIOptionsImpl;
+import org.hyperledger.besu.services.SecurityModuleServiceImpl;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import picocli.CommandLine;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LocalFileSecurityModulePluginTest {
+  @Rule public final TemporaryFolder temp = new TemporaryFolder();
+
+  @Mock private BesuContext besuContext;
+  private final SecurityModuleServiceImpl securityModuleService = new SecurityModuleServiceImpl();
+  private CommandLine commandLine;
+
+  @Before
+  public void setUp() {
+    commandLine = new CommandLine(TestCommand.class);
+
+    when(besuContext.getService(PicoCLIOptions.class))
+        .thenReturn(Optional.of(new PicoCLIOptionsImpl(commandLine)));
+    when(besuContext.getService(SecurityModuleService.class))
+        .thenReturn(Optional.of(securityModuleService));
+  }
+
+  @Before
+  @After
+  public void resetSystemProps() {
+    System.setProperty("besu.docker", "false");
+  }
+
+  @Test
+  public void cliOptionIsDisabledUnderDockerMode() {
+    System.setProperty("besu.docker", "true");
+    assumeTrue(Boolean.getBoolean("besu.docker"));
```

Comment:
Thinking an assertion would be better here, if the setProperty fails we want to know not just have the test be skipped.

Reply:
This one is adapted from the original unit test under BesuCommandTest. I believe assume would skip the test if condition fails (instead of failing the whole job).